go-controller/pkg/ovn/gateway_cleanup.go

```diff
@@ -19,14 +19,15 @@ import (
 )
 
 // gatewayCleanup removes all the NB DB objects created for a node's gateway
-func (oc *DefaultNetworkController) gatewayCleanup(nodeName string) error {
+// ignores any errors and continues with the cleanup if 'ignoreError' is set to true
+func (oc *DefaultNetworkController) gatewayCleanup(nodeName string, ignoreError bool) error {
```

Comment:
> I am not sure it is ok to ignore all errors because that could mean there might be leftover things that we should have deleted that would then never be deleted.

My latest update does continue with the cleanup even if one step fails.  I like the approach proposed by Dan to have an errors aggregator and the caller can take a call to ignore the errors or not.